Rename "paginator" option to "className" for consistency.

ADmad opened the pull request:
Also rename Paginator to DefaultPaginator for convenient usage of App::className().

<!---

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

The best way to propose a feature is to open an issue first and discuss your ideas there before implementing them.

Always follow the [contribution guidelines](https://github.com/cakephp/cakephp/blob/master/.github/CONTRIBUTING.md) when submitting a pull request. In particular, make sure existing tests still pass, and add tests for all new behavior. When fixing a bug, you may want to add a test to verify the fix.

-->

ADmad committed:
Rename "paginator" option to "className" for consistency.

Also rename Paginator to DefaultPaginator for convenient usage of App::className().

markstory commented on src/Controller/Component/PaginatorComponent.php:
```diff
@@ -34,15 +34,15 @@
  *
  * @link https://book.cakephp.org/4/en/controllers/components/pagination.html
- * @mixin \Cake\Datasource\Paging\Paginator
+ * @mixin \Cake\Datasource\Paging\DefaultPaginator
```
Are you open to calling this `BasicPaginator` instead? We used `Default` for password hashing in Authentication and that name adds a layer of indirection to finding out what the 'default' is. With 'basic' we can infer a bit more than we can with 'default'.

ADmad replied:
I am fine with renaming the class. But we already have a `SimplePaginator`, so `BasicPaginator` becomes too similar. Can we name the default class as `NumericPaginator` or `SlidingPaginator`? I am open to other suggestions too as long as we can easily distinguish the 2 paginators we have.

markstory replied:
🤦  I forgot we already had `SimplePaginator`. Totally agree that having both Simple and Basic is 🙃 . `NumericPaginator` works for me. It still leaves room for cursor and key based paginators in the future.

ADmad replied:
Renamed `DefaultPaginator` to `NumericPaginator`.

ADmad committed:
Rename "DefaultPaginator" to "NumericPaginator".

The former name adds a layer of indirection to finding out what the 'default' is.
